tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc

```diff
@@ -51,18 +51,19 @@ using GPUDevice = Eigen::GpuDevice;
 // size OuterDimTileSize x 1. This strip runs across multiple
 // rows of input data and all reduction elements share one inner
 // dimension index.
-template <typename T, typename Index, int OuterDimTileSize>
-__global__ void SortedSegmentSumCustomKernel(
+template <typename T, typename Index, int OuterDimTileSize,
+          typename SegmentReductionF, typename SegmentAtomicReductionF>
```

Comment:
WDYT about calling these `ReductionFunc` and `AtomicReductionFunc`?  There is nothing segment-specific to them.

Reply:
I do agree with you.